tests/unit/cmk/base/plugins/agent_based/utils/test_fileinfo_utils.py

```diff
@@ -420,11 +443,36 @@ def test__fileinfo_check_function(  # type:ignore[no-untyped-def]
                     summary="Conjunction: size at 12 B AND newest age below 1 day 0 hours",
                 )
             ],
-        )
+        ),
     ],
 )
 def test__fileinfo_check_conjunctions(  # type:ignore[no-untyped-def]
     check_definition, params, expected_result
 ) -> None:
     result = list(_fileinfo_check_conjunctions(check_definition, params))
     assert result == expected_result
+
+
+@pytest.mark.parametrize(
+    "check_definition, params",
+    [
+        (
+            [
+                ("Count", "count", 0, saveint),
+                ("Size", "size", 0, get_filesize_human_readable),
+                ("Largest size", "size_largest", None, get_filesize_human_readable),
+                ("Smallest size", "size_smallest", None, get_filesize_human_readable),
+                ("Oldest age", "age_oldest", None, get_age_human_readable),
+                ("Newest age", "age_newest", None, get_age_human_readable),
+            ],
+            {"conjunctions": [(2, [("age_oldest_lower", 129600)])], "maxcount": (10, 20)},
+        ),
+    ],
+)
+def test_typeerror__fileinfo_check_conjunctions(  # type:ignore[no-untyped-def]
+    check_definition, params
+) -> None:
+    try:
+        _result = list(_fileinfo_check_conjunctions(check_definition, params))
+    except TypeError:
+        pytest.fail("TypeError in _fileinfo_check_conjunctions with empty file list.")
```

Comment:
pytest has some handy helpers for this
```python
with pytest.raises(TypeError):
    list(_fileinfo_check_conjuncations(check_definition, params))
```

Reply:
with pytest.raises(TypeError)
expects a type error, right? But the unit test should pass, when NO TypeError is raised. I only found the solution with try and catch for this, as there ist no "pytest.notraises(TypeError)"

Reply:
So I now played around a bit with this. For cases where I do not want an exception to be thrown I usually do not use try/except. I just run the code in the try block and let any exception fail the test.